lib/puppet/util/windows/user.rb

```diff
@@ -63,8 +71,16 @@ def password_is?(name, password)
 
     begin
       logon_user(name, password) { |token| }
-    rescue Puppet::Util::Windows::Error
-      return false
+    rescue Puppet::Util::Windows::Error => detail
+
+      authenticated_error_codes = Set[
+        ERROR_ACCOUNT_RESTRICTION,
+        ERROR_INVALID_LOGON_HOURS,
+        ERROR_INVALID_WORKSTATION,
+        ERROR_ACCOUNT_DISABLED,
+      ]
+
+      return authenticated_error_codes.include?(detail.code)
```

Comment:
In my testing all of these came back with `1326` when triggered but with an invalid password.